Source/UI/Validator/PhoneNumberValidator.swift

```diff
@@ -29,11 +35,32 @@ class PhoneNumberValidator: Validator {
 
     func formatForDisplay(text: String) -> String {
         do {
-            let formattedNumber = try phoneKit.parse(text)
+            let formattedNumber = try phoneKit.parse(text,
+                                                     withRegion: countryCode)
             return phoneKit.format(formattedNumber, toType: .international)
         } catch {
             return text
         }
     }
 
 }
+
+// MARK: Singleton behaviour
+extension PhoneNumberValidator {
+
+    static var shared: PhoneNumberValidator {
+        if let instance {
+            return instance
+        }
+        let newValidator = PhoneNumberValidator()
+        instance = newValidator
+        return newValidator
```

Comment:
That would be nice, but `instance` is optional so it can be deallocated. 

Even though we just assign it the line before, compiler wouldn't know its been freshly created, it knows it may be nil. So unless we force unwrap and trigger the wrath of SwiftLint whilst risking future changes that will leave a bang operator hanging around, your suggestion would not work 😉